quack_app/test/screens/homepage/home_screen_test.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:golden_toolkit/golden_toolkit.dart';
+import 'package:quack_app/constants/constants.dart';
+import 'package:quack_app/core/item.dart';
+import 'package:quack_app/core/menu_data.dart';
+import 'package:quack_app/main.dart';
+import 'package:quack_app/screens/homepage/home_screen.dart';
+
+void main() {
+  group("Home Screen Test", () {
+    MenuData fakeMenuData = MenuData();
+    fakeMenuData.getCurrentlyServing();
+    fakeMenuData.getTodaysMenu();
+
+    MyApp app = const MyApp();
+    testGoldens("basic_view", (WidgetTester tester) async {
+      await loadAppFonts();
+      await tester.pumpWidget(app);
+
+      await expectLater(
+          find.byType(HomeScreen), matchesGoldenFile('goldens/basic_view.png'));
+    });
+
+    testGoldens("like_food", (WidgetTester tester) async {
+      await loadAppFonts();
+      await tester.pumpWidget(app);
+      await tester.tap(find.byIcon(Constants.kFavorite).first);
+      await tester.pumpAndSettle(const Duration(seconds: 2));
+      await expectLater(
+          find.byType(HomeScreen), matchesGoldenFile('goldens/like_food.png'));
+      int totalFavorited = 0;
+      for (Item item in fakeMenuData.menu) {
+        totalFavorited += item.isFavorite ? 1 : 0;
+      }
+      expect(totalFavorited, 1);
+    });
+
+    testGoldens("like_three_food", (WidgetTester tester) async {
```

Comment:
Not related to this PR, but I think we should override the WigetTester class so that the default resolution is closer to an actual device. Another approach is that we can make it test on multiple screen resolutions, which might be useful for finding UI bugs.

Reply:
I agree with your comments @DDavid233. Let's bring it up during story time so we can make it a ticket and get it on our TODO